### PR TITLE
Added Turf War Recipe and renumbered Binary Converter

### PR DESCRIPTION
**Pull request by codybsauer**
I worked on a version of this with a student during the level up camp and thought it might make a good recipe since the other kids enjoyed seeing it as well. It is originally inspired by the Nintendo game Splatoon where you try to paint/ink an area more than your opponents in a game mode called Turf War. If the result seems too simple or instructions are confusing/typo'd let me know.

### CODE REVIEW

File: Level1-Module1/src/_10_turf_war/TurfWar.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+        void drawPlayer() {
+            // 2. Pick a shape and draw the Player using its color, coordinates and size.
+            
+        }
+
+        void update() {
+            // Note: This piece of codes makes the player move up without leaving the bounds
+            // of the sketch if its up key is pressed.
+            if (moveUp && y > 0) {
```
Comment (dencee) [inline Level1-Module1/src/_10_turf_war/TurfWar.java:34]: Changed this to y > 0 from y>=0

File: Level1-Module1/src/_10_turf_war/TurfWar.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+//            }
+//
+//        } else if (player1.pixelCount > player2.pixelCount) {
+//            println("PLAYER 1 WINS");
+//        } else {
+//            println("PLAYER 2 WINS");
+//        }
+//        println((player1.pixelCount / (double) (width * height)) * 100 + "%");
+//        println((player2.pixelCount / (double) (width * height)) * 100 + "%");
+//        noLoop();
```
Comment (dencee) [inline Level1-Module1/src/_10_turf_war/TurfWar.java:146]: Added noLoop() to stop the game

**Review by dencee — COMMENTED** [review]
Cody, can you check that the conversion is good?

File: Level1-Module1/src/_10_turf_war/TurfWar.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+        // Note: These member variables do not need to be initialized in the constructor
+        boolean moveUp = false;
+        boolean moveDown = false;
+        boolean moveLeft = false;
+        boolean moveRight = false;
+        int pixelCount = 0;
+
+        void drawPlayer() {
+            // 2. Pick a shape and draw the Player using its color, coordinates and size.
```
Comment (dencee) [inline Level1-Module1/src/_10_turf_war/TurfWar.java:27]: can you specify which shape to draw? or if any shape is ok?

File: Level1-Module1/src/_10_turf_war/TurfWar.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        // control the Players without clicking on the window first.
+        ((java.awt.Canvas) surface.getNative()).requestFocus();
+
+        // 6. Set the background color.
+
+        // 7. Call the noStroke Method.
+
+        // 8. Initialize the two Player objects. For one use UP, LEFT, DOWN, RIGHT for
+        // the keys, for the second use the W,A,S,D int variables created above.
+        // Note: Use the Color class for the color, e.g. Color.BLUE.getRGB()
```
Comment (dencee) [inline Level1-Module1/src/_10_turf_war/TurfWar.java:114]: Using the Color class and getRGB() method to get an int.

**Review by dencee — COMMENTED** [review]
If the game had a timer that would count down to when the game ends that would be cool.

File: Level1-Module1/src/_10_turf_war/TurfWar.java (hunk 3 above)
Comment (codybsauer) [reply Level1-Module1/src/_10_turf_war/TurfWar.java:27]: A rectangle is probably best for the collision so I'll specify that.

**Comment by codybsauer** [comment]
> If the game had a timer that would count down to when the game ends that would be cool.

Worked on this and showing the active percentage of coverage during the game. Still have to test a couple things, but should get back to this on Sunday.

**Comment by codybsauer** [comment]
> Cody, can you check that the conversion is good?

Conversion works well. I just need to make the rest of my changes and this should be good.

File: Level1-Module1/src/_10_turf_war/TurfWar.java [hunk 5]
```diff
@@ -16,29 +21,47 @@
...
-                y--;
+            /*
+             * This piece of codes makes the player move up without
+             * leaving the bounds of the sketch if its up key is pressed.
+             * 
+             * Note: statsBoardLine is used instead of 0 because that is 
+             * where the area detailing game stats(timer, score, etc.) ends.
+             * 
+             */     
+            if (moveUp && y > statsBoardLine) {
```
Comment (dencee) [inline Level1-Module1/src/_10_turf_war/TurfWar.java:51]: should it be:
&& (y - speed) > statsBoardLine) {

Comment (dencee) [reply Level1-Module1/src/_10_turf_war/TurfWar.java:51]: Actually,
if (moveUp && y < statsBoardLine) {
    y = statsBoardLine + speed;
}

meh, doesn't really matter too much

**Review by dencee — APPROVED** [review]

**Pull request merged**